service/controller/clusterapi/v19/resources/app/desired.go

```diff
@@ -0,0 +1,96 @@
+package app
+
+import (
+	"context"
+	"strconv"
+
+	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/annotation"
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/pkg/project"
+	pkgkey "github.com/giantswarm/cluster-operator/pkg/v19/key"
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v19/key"
+	azurekey "github.com/giantswarm/cluster-operator/service/controller/azure/v19/key"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v19/key"
+	kvmkey "github.com/giantswarm/cluster-operator/service/controller/kvm/v19/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8sv1alpha1.App, error) {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var apps []*g8sv1alpha1.App
+
+	for _, appSpec := range r.newAppSpecs() {
+		apps = append(apps, r.newApp(cr, appSpec))
+	}
+
+	return apps, nil
+}
+
+func (r *Resource) newApp(cr cmav1alpha1.Cluster, appSpec pkgkey.AppSpec) *g8sv1alpha1.App {
+	return &g8sv1alpha1.App{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ChartConfig",
```

Comment:
This should be,
```suggestion
			Kind:       "App",
```
But you can also drop this `TypeMeta` in general.

Reply:
Oof 🤦‍♂  Yes I'll drop this.